packages/runtime-node/src/communication-helpers.ts

```diff
@@ -0,0 +1,65 @@
+import { ChildProcess, Serializable } from 'node:child_process';
+import { type MessagePort, type Worker } from 'node:worker_threads';
+import { deferred } from 'promise-assist';
+
+/**
+ * Emits an event using specified emitter as MessagePort
+ */
+export function emitEvent<T extends Serializable>(
```

Comment:
this shouldn't be generic. it's an anti-pattern to use it in a single fn parameter.
instead, caller should use `satisfies` on value being passed to this fn
and it should accept `event: unknown`

Reply:
removed these helpers